builtin/logical/aws/rotation_test.go

```diff
@@ -0,0 +1,307 @@
+	"github.com/hashicorp/vault/sdk/queue"
+)
+
+// TestRotation verifies that the rotation code and priority queue correctly selects and rotates credentials
+// for static secrets.
+func TestRotation(t *testing.T) {
+	bgCTX := context.Background()
+
+	type credToInsert struct {
+		config staticRoleEntry // role configuration from a normal createRole request
+		age    time.Duration   // how old the cred should be - if this is longer than the config.RotationPeriod,
+		// the cred is 'pre-expired'
+
+		changed bool // whether we expect the cred to change - this is technically redundant to a comparison between
+		// rotationPeriod and age.
+	}
+
+	cases := []struct {
+		name  string
+		creds []credToInsert
+	}{
+		{
+			name: "refresh one",
+			creds: []credToInsert{
+				{
+					config: staticRoleEntry{
+						Name:           "test",
+						Username:       "jane-doe",
+						RotationPeriod: 2 * time.Second,
+					},
+					age:     5 * time.Second,
+					changed: true,
+				},
+			},
+		},
+		{
+			name: "refresh none",
+			creds: []credToInsert{
+				{
+					config: staticRoleEntry{
+						Name:           "test",
+						Username:       "jane-doe",
+						RotationPeriod: 1 * time.Minute,
+					},
+					age:     5 * time.Second,
+					changed: false,
+				},
+			},
+		},
+		{
+			name: "refresh one of two",
+			creds: []credToInsert{
+				{
+					config: staticRoleEntry{
+						Name:           "test",
+						Username:       "jane-doe",
+						RotationPeriod: 1 * time.Minute,
+					},
+					age:     5 * time.Second,
+					changed: false,
+				},
+				{
+					config: staticRoleEntry{
+						Name:           "toast",
+						Username:       "john-doe",
+						RotationPeriod: 1 * time.Second,
+					},
+					age:     5 * time.Second,
+					changed: true,
+				},
+			},
+		},
+		{
+			name:  "no creds to rotate",
+			creds: []credToInsert{},
+		},
+	}
+
+	ak := "long-access-key-id"
+	oldSecret := "abcdefghijklmnopqrstuvwxyz"
+	newSecret := "zyxwvutsrqponmlkjihgfedcba"
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			config := logical.TestBackendConfig()
+			config.StorageView = &logical.InmemStorage{}
+
+			b := Backend()
+			// this means the creds will be the same for every user, but that's okay
+			// since what we care about is whether they changed on a single-user basis.
+			miam, err := awsutil.NewMockIAM(
+				// blank list for existing user
+				awsutil.WithListAccessKeysOutput(&iam.ListAccessKeysOutput{
+					AccessKeyMetadata: []*iam.AccessKeyMetadata{
+						{},
+					},
+				}),
+				// initial key to store
+				awsutil.WithCreateAccessKeyOutput(&iam.CreateAccessKeyOutput{
+					AccessKey: &iam.AccessKey{
+						AccessKeyId:     aws.String(ak),
+						SecretAccessKey: aws.String(oldSecret),
+					},
+				}),
+			)(nil)
+			if err != nil {
+				t.Fatalf("couldn't initialze mock IAM handler: %s", err)
+			}
+
+			// insert all our creds
+			for i, cred := range c.creds {
+				b.iamClient = miam
```

Comment:
nit: doesn't look like we'll be changing `b.iamClient` between runs of the loop; should we set this outside the loop as done below in [L165](https://github.com/hashicorp/vault/pull/20536/files#diff-118244014238638c5f6c27281d9f4978a27baba6d33308b5275e8a0dd1a0b90cR165)?